src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -742,9 +724,22 @@ int MPIDI_OFI_init_local(int *tag_bits)
     mpi_errno = MPIDI_OFI_init_multi_nic(prov);
     MPIR_ERR_CHECK(mpi_errno);
 
-    for (int i = 0; i < MPIDI_OFI_global.num_nics; i++) {
-        /* if MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS, set rx_ctx_cnt and tx_ctx_cnt */
-        set_sep_counters(i);
+    if (MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS) {
+#ifdef MPIDI_OFI_VNI_USE_DOMAIN
+        /* Note: currently we request a single tx and rx ctx under MPIDI_OFI_VNI_USE_DOMAIN */
+        int num_ctx_per_nic = 1;
+#else
+        /* FIXME: use max_vcis */
+        int num_ctx_per_nic = MPIDI_OFI_global.num_vcis;
```

Comment:
What is impacted when changing this from `MPIR_CVAR_CH4_NUM_VCIS + MPIR_CVAR_CH4_RESERVE_VCIS` to `MPIDI_OFI_global.num_vcis`?

Reply:
Good catch! I will change it back. At the time I was thinking to make `num_vcis` on-demand, so it can be larger than `MPIR_CVAR_CH4_NUM_VCIS + MPIR_CVAR_CH4_RESERVE_VCIS`. But alas that won't work since this is at the early stage of init.

As for impact, using scalable endpoints for vci has been broken for a while 🥲

Reply:
Fixed.